python/ray/data/aggregate.py

```diff
@@ -1,5 +1,13 @@
+import functools
 import math
-from typing import Callable, Optional, List, TYPE_CHECKING
+from typing import Callable, Optional, List, TYPE_CHECKING, Any
+
+try:
+    import pandas as pd
```

Comment:
We shouldn't eager import pandas. This has caused issues in the past. Use lazy import pattern instead: https://github.com/ray-project/ray/blob/master/python/ray/data/impl/pandas_block.py#L53

Reply:
Ah that's right, forgot about those issues. I'll port to the lazy import pattern!